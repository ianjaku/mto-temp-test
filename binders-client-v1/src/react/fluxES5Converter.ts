/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
