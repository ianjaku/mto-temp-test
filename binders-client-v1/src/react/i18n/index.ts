import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
