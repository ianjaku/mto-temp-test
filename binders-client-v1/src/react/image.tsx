/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import { pick, replace } from "ramda";
import { isVideoURL } from "../clients/imageservice/v1/visuals";

export type Unit = "%" | "vw";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function getFitCropStyleObject(fitBehaviour: string, aspectRatio: number, unit: Unit = "%") {
    const isLandscape = aspectRatio > 1;
    const addUnit = (amount: number) => `${amount}${unit}`;
    if (fitBehaviour === "fit") {
        if (isLandscape) {
            return {
                width: addUnit(100)
            };
        } else {
            const style = {
                height: addUnit(100),
                width: addUnit(Math.round(100 * aspectRatio))
            };
            return style;
        }
    } else {
        if (isLandscape) {
            return {
                left: addUnit(50),
                top: addUnit(50),
                height: addUnit(100),
                width: "auto",
                transform: `translate(-${addUnit(50)},-${addUnit(50)})`,
                position: "absolute"
            };
        } else {
            return {
                width: addUnit(100)
            };
        }
    }
}


export interface ImageProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    image?: any;
    isOver?: boolean;
    dirtyDownSize?: boolean;
    className: string;
}

export interface ImageState {
    bgColor: string;
    width?: number;
    height?: number;
}

export default class Image extends React.Component<ImageProps, ImageState> {

    constructor(props) {
        super(props);
        this.state = {
            bgColor: props.image ? props.image.bgColor : "",
        };
    }

    handleImageLoaded(e) {
        this.setState({
            bgColor: this.props.image.bgColor,
            width: e.target.naturalWidth,
            height: e.target.naturalHeight
        });
    }

    dirtyDownsizeImageImg(img) {
        let url;
        if (img.url) {
            url = img.url;
        } else {
            url = img;
        }
        let replaceTarget = "/medium";
        if (isVideoURL(url)) {
            replaceTarget = "/video_screenshot";
        }
        return replace(/\/original/, replaceTarget, url);
    }

    getImageSrc() {
        if (this.props.dirtyDownSize) {
            return this.dirtyDownsizeImageImg(this.props.image);
        }
        if (this.props.image.buildRenderUrl) {
            return this.props.image.buildRenderUrl();
        }
        if (this.props.image.medium) {
            return this.props.image.medium;
        }
        return this.props.image;
    }

    getImageProps() {
        const { rotation }  = this.props.image;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const imageProps: any = {
            ...pick(["style", "alt"], this.props),
            className: this.props.isOver ? "hovering" : "",
            src: this.getImageSrc(),
            onLoad: this.handleImageLoaded.bind(this)
        };
        let rotationStyle = {};
        imageProps.style = Object.assign(imageProps.style ? imageProps.style : {}, this.getFitCropStyling());
        if(rotation) {
            const rotate = `rotate(${rotation}deg)`;
            rotationStyle = {transform:  (imageProps.style.transform) ? `${imageProps.style.transform} ${rotate}` : rotate};
            imageProps.style = {...imageProps.style, ...rotationStyle};
        }
        return imageProps;
    }

    getFitCropStyling() {
        return getFitCropStyleObject(this.props.image.fitBehaviour, this.getAspectRatio());
    }

    getAspectRatio() {
        return this.state.height ?
            this.state.width / this.state.height :
            1;
    }

    render() {
        const isTransparent = this.props.image.bgColor === "transparent";
        const wrapperStyle = { backgroundColor: isTransparent ? "inherit" : `#${this.props.image.bgColor}` } as React.CSSProperties;

        return (
            <div className={`${this.props.className ? this.props.className : ""} image-wrapper ${(isTransparent ? " transparent" : "")}`} style={wrapperStyle}>
                <div className="image-container">
                    <img {...this.getImageProps() } />
                </div>
            </div>
        );
    }

}
