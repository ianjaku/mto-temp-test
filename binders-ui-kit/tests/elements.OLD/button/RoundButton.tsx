import * as React from "react";
import AddChunkButton from "../../../src/elements/button/AddChunk";
import MergeChunkButton from "../../../src/elements/button/MergeChunk";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import wrapWithTheme from "../../themeHelper";

const createMergeChunkButton = callback => wrapWithTheme(<MergeChunkButton onClick={callback} />);
const createAddChunkButton = callback => wrapWithTheme(<AddChunkButton onClick={callback} />);

describe("Rounded button", () => {
    test("Merge chunk button (snapshot)", () => {
        const modal = create(createMergeChunkButton(undefined));
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("Add chunk button (snapshot)", () => {
        const modal = create(createAddChunkButton(undefined));
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("Merge chunk button (functional)", () => {
        const callback = jest.fn();

        const dom = mount(createMergeChunkButton(callback));
        const button = dom.find(".button-merge-chunks").at(0);
        button.simulate("click");
        expect(callback.mock.calls.length).toEqual(1);
    });

    test("Add chunk button (functional)", () => {
        const callback = jest.fn();

        const dom = mount(createAddChunkButton(callback));
        const button = dom.find(".button-add-chunk").at(0);
        button.simulate("click");
        expect(callback.mock.calls.length).toEqual(1);
    });
});
