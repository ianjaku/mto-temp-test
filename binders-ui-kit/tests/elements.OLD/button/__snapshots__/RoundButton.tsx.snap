// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rounded button Add chunk button (snapshot) 1`] = `
<button
  className="button-round button-add-chunk"
  onClick={undefined}
>
  <span
    className="material-icons "
    color=""
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fontSize": 24,
        "muiPrepared": true,
        "position": "relative",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "userSelect": "none",
      }
    }
  >
    add
  </span>
</button>
`;

exports[`Rounded button Merge chunk button (snapshot) 1`] = `
<button
  className="button-round button-merge-chunks"
  onClick={undefined}
>
  <span
    className="material-icons "
    color=""
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fontSize": 24,
        "muiPrepared": true,
        "position": "relative",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "userSelect": "none",
      }
    }
  >
    vertical_align_center
  </span>
</button>
`;
